filestore.go

```diff
@@ -117,53 +118,46 @@ func (f *Filestore) AllKeysChan(ctx context.Context) (<-chan cid.Cid, error) {
 // ErrNotFound when the block is not stored.
 func (f *Filestore) DeleteBlock(ctx context.Context, c cid.Cid) error {
 	err1 := f.bs.DeleteBlock(ctx, c)
-	if err1 != nil && err1 != blockstore.ErrNotFound {
+	if err1 != nil && !ipld.IsNotFound(err1) {
 		return err1
 	}
 
 	err2 := f.fm.DeleteBlock(ctx, c)
 	// if we successfully removed something from the blockstore, but the
 	// filestore didnt have it, return success
 
-	switch err2 {
-	case nil:
-		return nil
-	case blockstore.ErrNotFound:
-		if err1 == blockstore.ErrNotFound {
-			return blockstore.ErrNotFound
+	if ipld.IsNotFound(err2) {
+		if ipld.IsNotFound(err1) {
+			return err1
 		}
+		// being here means err1 was nil and err2 NotFound.
 		return nil
-	default:
-		return err2
 	}
+
+	return err2
```

Comment:
```suggestion
	if !ipld.IsNotFound(err2) {
		return err2
	}
	
	if ipld.IsNotFound(err1) {
		return err1
	}
	
	return nil
```

This functionally identical, but clearer to me (I really had issues understanding your code, the `return nil` mixed between both cases made it hard on my brain for some reason).